plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeMetadata.java

```diff
@@ -885,6 +886,15 @@ public Optional<ConnectorOutputMetadata> finishCreateTable(
             metastore.createTable(session, table, principalPrivileges);
         }
         catch (Exception e) {
+            Optional<Table> existingTable = metastore.getTable(schemaName, tableName);
```

Comment:
`io.trino.plugin.hive.TableAlreadyExistsException` is thrown. Moved the block to catch only exceptions from `metastore.createTable()`.

Reply:
@losipiuk @ebyhr Maybe we should handle this issue internally in Metastore client.
I mean it already happened several times and maybe with create operations it would be worth to handle it on client lvl instead of propagate error which is not an error at the end ?

Reply:
This is a good point. The question is can we (at a metastore level) determine if the table which appeared in MS is result of retry vs. race condition with some other caller.

Reply:
I approached this some time ago. It may be determined.
When we perform call over Thrift with any crate operation we receive ThriftException with clear message "read timeout". At that point we could just check if table exists instead of retrying. If it does not exists we failed if not it was just read timeout.

Reply:
Nope - if it exists it may still be that someone else created it. We should check if the table which exists is the table we did in previous retry invocation - we did that by checking query-id from metadata in this PR.